### ISSUE TITLE
aws_codedeploy_deployment_group should support tag groups

### ISSUE THREAD

**Issue by davedash** [terraform-providers/terraform-provider-aws#2073]
Tag groups aren't supported in `aws_codedeploy_deployment_group`.  This will allow you do intersections of tags.

<img width="431" alt="screen shot 2017-10-26 at 1 03 29 pm" src="https://user-images.githubusercontent.com/59497/32074392-1cc3acc2-ba4e-11e7-9ab2-7ac87b4f2bf6.png">

**Comment by rmenn** [comment]
IIRC this should do it for you - https://www.terraform.io/docs/providers/aws/r/codedeploy_deployment_group.html#ec2_tag_filter

**Comment by davedash** [comment]
@rmenn the ec2_tag_filter places the tags into a single tag group (from my experiments) and according to code deploy that means it will deploy to things matching ANY of the tags.

So in the above example it would make a single tag group with env=dev, role=rails and match all my instances matching dev and all the instances matching rails.  Rather than just instances matching dev *and* rails.

**Comment by davedash** [comment]
FYI, for the work around, I created the tag groups manually and Terraform ignores them, which is fine

**Comment by logicbomb421** [comment]
This is also affecting my workflow. Seems to be a pretty big oversight for the CodeDeploy resource. Any chance we can get an ETA on a fix?

**Comment by jaredready** [comment]
Bumping this issue. As we're currently just using Terraform for our CodeDeploy configuration and the associated servers, this is a big deal. We can workaround it by changing our tag scheme, but we would much rather have the flexibility of using tag groups.

**Comment by logicbomb421** [comment]
Current workaround I'm using is to use the `local-ecex` provisioner and the aws-cli to create these tag sets:
```hcl
  provisioner "local-exec" {
    command = <<EOF
    aws deploy update-deployment-group \
      --application-name ${aws_codedeploy_deployment_group.main.app_name} \
      --current-deployment-group-name ${aws_codedeploy_deployment_group.main.deployment_group_name} \
      --ec-2-tag-set '{
          "ec2TagSetList": [
            [
              {
                "Key": "Name",
                "Value": "${var.app_name}",
                "Type": "KEY_AND_VALUE"
              }
            ],
            [
              {
                "Key": "Environment",
                "Value": "${var.environment}",
                "Type": "KEY_AND_VALUE"
              }
            ]
          ]
        }'
    EOF
  }
```

**Comment by bflad** [comment]
The `ec2_tag_set` argument has been added to the `aws_codedeploy_deployment_group` attribute via #4324 and will release with v1.17.0 of the AWS provider, likely in a week.

**Comment by bflad** [comment]
This has been released in [version 1.17.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1170-may-02-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
